fdb-relational-core/src/main/java/com/apple/foundationdb/relational/recordlayer/RecordLayerIterator.java

```diff
@@ -83,6 +83,7 @@ public boolean hasNext() {
 
     private void fetchNextResult() {
         if (result != null) {
+            continuation = ContinuationImpl.fromRecordCursorContinuation(result.getContinuation());
```

Comment:
Hm, I'm not sure this is quite right. It seems like there is perhaps a deficiency in the previous code, but perhaps it should set the continuation on below near where it sets the continuation if the `noNextReason` is `SOURCE_EXHAUSTED`. That is, instead of just setting it to `END` if the no next reason is `SOURCE_EXHAUSTED`, it should also set it to the copied over continuation if the no-next-reason is not `SOURCE_EXHAUSTED`

Reply:
you're right, changed.